### PR TITLE
[DOC] Formatting of docstring in regions.Parcellations

### LINKED ISSUE #3914: [DOC] Formatting of docstring in regions.Parcellations
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed suggestion in detail.

Right now, the line break in the docstring for `nilearn.regions.Parcellations` is causing sub-optimal formatting.

Here's the rendered docstring
Screenshot 2023-08-07 at 20 20 56

and the accompanying docstring:
https://github.com/nilearn/nilearn/blob/321494420f95c7a5e2172108400194b37a02e628/nilearn/regions/parcellations.py#L135-L136

As you can see, "hierarchical_kmeans" is not formatted in the same styling as the other valid methods.

### List any pages that would be impacted.

https://nilearn.github.io/stable/modules/generated/nilearn.regions.Parcellations.html#nilearn.regions.Parcellations

### PR DESCRIPTION
**Pull request by NIkhilgKrish**
- Closes #3914 

Changes proposed in this pull request:

- Add backslash to homogenize nilearn.regions.Parcellations.html documentation

### CODE REVIEW

File: nilearn/regions/parcellations.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ class Parcellations(_MultiPCA):
 
     Parameters
     ----------
-    method: :obj:`str`, {'kmeans', 'ward', 'complete', 'average', 'rena',
+    method: :obj:`str`, {'kmeans', 'ward', 'complete', 'average', 'rena', \
```
[the remark this answers is not in the document]
Comment (NIkhilgKrish) [reply nilearn/regions/parcellations.py:136]: Certainly. I will submit the revision after making the changes

File: doc/changes/latest.rst [hunk 2]
```diff
@@ -19,6 +19,8 @@ Fixes
 Enhancements
 ------------
 
+- :bdg-primary:`Doc` Add backslash to homogenize nilearn.regions.Parcellations.html documentation (:gh:`4042` by `Nikhil Krish`_).
```
Comment (Remi-Gau) [inline doc/changes/latest.rst:22]:
```suggestion
- :bdg-primary:`Doc` Add backslash to homogenize :class:`~nilearn.regions.Parcellations` documentation (:gh:`4042` by `Nikhil Krish`_).
```

**Review by bthirion — APPROVED** [review]
LGTM, thx!

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
thanks @NIkhilgKrish

**Pull request merged**